### PR TITLE
Add surface area price ceiling calculation data to database for reporting

### PR DESCRIPTION
**Pull request by matti-lamppu**
# Hitas Pull Request

# Description

Adds the ability to save surface area price ceiling calculation data to the database. The saved data may not be final and can be changed once the reports based on it are made.

Also, try to fix a flaky test in thirty-year regulation.

## Pull request checklist

Check the boxes for each DoD item that has been completed:

- **Testing**
  - [x] Changes have been tested
  - [x] Automatic tests have been added
- **Database**
  - [ ] Database migrations will work in the DEV & TEST environments
  - [ ] initial.json has been updated to work with migrations
  - [ ] Oracle migration has been updated
- **Documentation**
  - [ ] Tooltips have been added in the frontend for all new fields
  - [ ] OpenAPI definitions have been updated
  - [ ] Test instructions have been written for the customer in the appropriate ticket in Jira
  - [ ] Terminology page in Confluence has been updated

## Test plan

- [x] Automated tests
- [x] Make a surface area price ceiling calculation and look at the database for calculation results

## Tickets

This pull request resolves all or part of the following ticket(s): HT-434

### CODE REVIEW

File: backend/hitas/tests/apis/test_api_thirty_year_regulation.py [hunk 1]
```diff
@@ -511,6 +511,13 @@ def test__api__regulation__automatically_release__partial(api_client: HitasAPICl
         apartment__building__real_estate__housing_company__state=HousingCompanyState.LESS_THAN_30_YEARS,
     )
 
+    # For some reason, financing method old ruleset flag may not always get set with the factory,
+    # resulting in the test being flaky, so try to fix the flakiness with explicit setting
+    sale_1.apartment.housing_company.financing_method.old_hitas_ruleset = False
+    sale_1.apartment.housing_company.financing_method.save()
+    sale_2.apartment.housing_company.financing_method.old_hitas_ruleset = True
+    sale_2.apartment.housing_company.financing_method.save()
+
```
Comment (ranta) [inline backend/hitas/tests/apis/test_api_thirty_year_regulation.py:520]: Fix and explanation for this test is already included in 431e5a6e48722fa797aa49ce687c492dd22e7655 :)

File: backend/hitas/migrations/0063_surfaceareapriceceilingcalculationdata.py
Comment (ranta) [inline backend/hitas/migrations/0063_surfaceareapriceceilingcalculationdata.py:1]: Migration needs to be renamed to 0064

File: backend/hitas/tests/apis/test_api_thirty_year_regulation.py (hunk 1 above)
Comment (matti-lamppu) [reply backend/hitas/tests/apis/test_api_thirty_year_regulation.py]: Yeah, dropped the commit.

File: backend/hitas/migrations/0063_surfaceareapriceceilingcalculationdata.py
Comment (matti-lamppu) [reply backend/hitas/migrations/0063_surfaceareapriceceilingcalculationdata.py:1]: Done

**Review by ranta — APPROVED** [review]
🦖

**Pull request merged**